lib/js/src/rpc/structs/ButtonCapabilities.js

```diff
@@ -0,0 +1,138 @@
+/*
+* Copyright (c) 2019, Livio, Inc.
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* Redistributions of source code must retain the above copyright notice, this
+* list of conditions and the following disclaimer.
+*
+* Redistributions in binary form must reproduce the above copyright notice,
+* this list of conditions and the following
+* disclaimer in the documentation and/or other materials provided with the
+* distribution.
+*
+* Neither the name of the Livio Inc. nor the names of its contributors
+* may be used to endorse or promote products derived from this software
+* without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+* AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+* IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+* ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+* LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+* CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+
+import { RpcStruct } from '../RpcStruct.js';
+import { ModuleInfo } from './ModuleInfo';
+import { ButtonName } from '../enums/ButtonName.js';
+
+class ButtonCapabilities extends RpcStruct {
+
+    constructor(parameters) {
+        super(parameters);
+    }
+
+    /**
+    * @param {ButtonName} name
+    * @return {ButtonCapabilities}
+    */
+    setName(name) {
+        this.validateType(ButtonName, name);
+
+        this.setValue(ButtonCapabilities.KEY_NAME, name);
+        return this;
+    }
+
+    /**
+    * @return {ButtonName}
+    */
+    getName() {
+        return this.getObject(ButtonCapabilities, ButtonCapabilities.KEY_NAME);
+    }
+
+    /**
+    * @param {ModuleInfo} moduleInfo
+    * @return {ButtonCapabilities}
+    */
+    setModuleInfo(moduleInfo) {
+        this.validateType(ModuleInfo, moduleInfo);
+
+        this.setValue(ButtonCapabilities.KEY_MODULE_INFO, moduleInfo);
+        return this;
+    }
+
+    /**
+    * @return {ModuleInfo}
+    */
+    getModuleInfo() {
+        return this.getObject(ModuleInfo, ButtonCapabilities.KEY_MODULE_INFO);
+    }
+
+    /**
+    * @param {Boolean} shortPressAvailable
+    * @return {ButtonCapabilities}
+    */
+    setShortPressAvailable(shortPressAvailable) {
+
+        this.setValue(ButtonCapabilities.KEY_SHORT_PRESS_AVAILABLE, shortPressAvailable);
```

Comment:
`shortPressAvailable` param is simple Boolean, shouldn't we use `this.setParameter(ButtonCapabilities.KEY_SHORT_PRESS_AVAILABLE, shortPressAvailable);` instead of `setValue`?